yellowbrick/contrib/missing/bar.py

```diff
@@ -0,0 +1,247 @@
+
+
+class MissingValuesBar(MissingDataVisualizer):
+    """The MissingValues Bar visualizer creates a bar graph that lists the total
+    count of missing values for each selected feature column.
+
+    When y targets are supplied to fit, the output is a stacked bar chart where
+    each color corresponds to the total NaNs for the feature in that column.
+
+    Parameters
+    ----------
+    alpha : float, default: 0.5
+        A value for bending elments with the background.
+
+    marker : matplotlib marker, default: |
+        The marker used for each element coordinate in the plot
+
+    color : string, default: black
+        The color for drawing the bar chart when the y targets are not passed to
+        fit.
+
+    colors : list, default: None
+        The color pallette for drawing a stack bar chart when the y targets
+        are passed to fit.
+
+    classes : list, default: None
+        A list of class names for the legend.
+        If classes is None and a y value is passed to fit then the classes
+        are selected from the target vector.
+
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+
+    Attributes
+    ----------
+    features_ : np.array
+        The feature labels ranked according to their importance
+
+    classes_ : np.array
+        The class labels for each of the target values
+
+    Examples
+    --------
+
+    >>> from yellowbrick.contrib.missing import MissingValuesBar
+    >>> visualizer = MissingValuesBar()
+    >>> visualizer.fit(X, y=y)
+    >>> visualizer.poof()
+    """
+
+    def __init__(self, width=0.5, color='black', colors=None, classes=None, **kwargs):
+        super(MissingValuesBar, self).__init__(**kwargs)
+        self.width = width  # the width of the bars
+        self.classes_ = classes
+        self.ind = None
+
+        # Convert to array if necessary to match estimator.classes_
+        if self.classes_ is not None:
+            self.classes_ = np.array(classes)
+
+        # Set up classifier score visualization properties
+        if self.classes_ is not None:
+            n_colors = len(self.classes_)
+        else:
+            n_colors = None
+
+        self.colors = color_palette(kwargs.pop('colors', None), n_colors)
+
+    def get_nan_col_counts(self, **kwargs):
+        # where matrix contains strings, handle them
+        if np.issubdtype(self.X.dtype, np.string_) or np.issubdtype(self.X.dtype, np.unicode_):
+            mask = np.where( self.X == '' )
+            nan_matrix = np.zeros(self.X.shape)
+            nan_matrix[mask] = np.nan
+
+        else:
+            nan_matrix = self.X.astype(np.float)
+
+        if self.y is None:
+            nan_col_counts = [np.count_nonzero(np.isnan(col)) for col in nan_matrix.T]
+            return nan_col_counts
+
+        else:
+            # add in counting of np.nan per target y by column
+            nan_counts = []
```

Comment:
This array seems like something the user would be interested in, e.g. stored in `self.nan_counts_` since it's a learned parameter. It would work for both `nan_col_counts` and `nan_counts` - scikit-learn routinely changes the shape of its underlying learned parameters depending on the input to the transformer.